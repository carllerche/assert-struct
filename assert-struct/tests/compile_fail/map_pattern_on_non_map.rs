use assert_struct::assert_struct;

#[derive(Debug)]
struct User {
    name: String,
    age: u32,
}

fn main() {
    let user = User {
        name: "Alice".to_string(),
        age: 30,
    };

    // This should NOT compile - User doesn't have len() and get() methods
    assert_struct!(user, User {
        name: #{ "key": "value" },
        age: 30,
    });
}