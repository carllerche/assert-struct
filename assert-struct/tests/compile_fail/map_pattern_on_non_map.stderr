error[E0277]: the type `str` cannot be indexed by `&String`
  --> tests/compile_fail/map_pattern_on_non_map.rs:17:18
   |
17 |         name: #{ "key": "value" },
   |                  ^^^^^ string indices are ranges of `usize`
   |
   = help: the trait `SliceIndex<str>` is not implemented for `&String`
note: required by a bound in `core::str::<impl str>::get`
  --> $RUST/core/src/str/mod.rs
